commcare_connect/opportunity/views.py

```diff
@@ -1046,3 +1048,49 @@ def user_visit_review(request, org_slug, opp_id):
         "opportunity/user_visit_review.html",
         context=dict(table=table, user_visit_ids=[v.pk for v in user_visit_reviews]),
     )
+
+
+@org_member_required
+def payment_report(request, org_slug, pk):
+    opportunity = get_opportunity_or_404(pk, org_slug)
+    if not opportunity.managed:
+        return redirect("opportunity:detail", org_slug, pk)
+    payment_units = PaymentUnit.objects.filter(opportunity=opportunity)
+    PaymentReportData = namedtuple(
+        "PaymentReportData", ["payment_unit", "approved", "user_payment_accrued", "nm_payment_accrued"]
+    )
+    data = []
+    total_paid_users = (
```

Comment:
nit: the queries may be cleaner in a helper function like the other big complex tables. I know those were single db queries, but i think its fine to have complex ones like this there too, becuase it separates it from the view code, making the view easier to read.

Reply:
Fixed. a02109cce56287b6d6d6806e928542905a5f7626